test/jdk/java/net/httpclient/HttpResponseConnectionLabelTest.java

```diff
@@ -0,0 +1,444 @@
+            URI requestUri = URI.create(requestUriScheme + "://" + server.serverAuthority() + handlerPath);
+
+            // Register the request handler
+            String serverId = "" + SERVER_COUNTER.getAndIncrement();
+            CountDownLatch[] serverResponseLatchRef = new CountDownLatch[1];
+            server.addHandler(
+                    (exchange) -> {
+                        String responseBody = "" + SERVER_RESPONSE_COUNTER.getAndIncrement();
+                        String connectionKey = exchange.getConnectionKey();
+                        try (exchange) {
+
+                            // Participate in the latch count down
+                            if (serverResponseLatchRef[0] != null) {
+                                serverResponseLatchRef[0].countDown();
+                                LOGGER.log(
+                                        "Server[%s] is waiting for the latch... (connectionKey=%s, responseBody=%s)",
+                                        serverId, connectionKey, responseBody);
+                                serverResponseLatchRef[0].await();
+                            }
+
+                            // Write the response
+                            LOGGER.log(
+                                    "Server[%s] is responding... (connectionKey=%s, responseBody=%s)",
+                                    serverId, connectionKey, responseBody);
+                            exchange.getResponseHeaders().addHeader(CONNECTION_KEY_HEADER_NAME, connectionKey);
+                            exchange.getResponseHeaders().addHeader(SERVER_ID_HEADER_NAME, serverId);
+                            byte[] responseBodyBytes = responseBody.getBytes(CHARSET);
+                            exchange.sendResponseHeaders(200, responseBodyBytes.length);
+                            exchange.getResponseBody().write(responseBodyBytes);
+
+                        } catch (IOException ioe) {
+                            String message = "Server[%s] has failed! (connectionKey=%s, responseBody=%s)"
+                                    .formatted(serverId, connectionKey, responseBody);
+                            throw new RuntimeException(message, ioe);
+                        } catch (InterruptedException _) {
+                            Thread.currentThread().interrupt(); // Restore the interrupt
+                        }
+                    },
+                    handlerPath);
+
+            // Create the client and the request
+            HttpRequest request = HttpRequest.newBuilder(requestUri).version(version).build();
+
+            // Create the pair
+            return new ServerRequestPair(server, request, secure, serverResponseLatchRef);
+
+        }
+
+        private static HttpTestServer createServer(Version version, SSLContext sslContext) {
+            try {
+                return HttpTestServer.create(version, sslContext, ForkJoinPool.commonPool());
```

Comment:
OK - we definitely MUST NOT use the common FJP here. 
1. our HTTP/2 server start their dispatching thread, which blocks on accept(), in the executor. 
2. the tests will block (CountDownLatch) in the exchange, which is a thread in the executor.
If you run this test on a machine that has a low number of processors - say 4 - then the FJP will maybe have 3 threads at most. You can see where that goes...

Reply:
If no executor is provide the HTTP/2 server will create its own. The HTTP/1.1 server howevr will run the exchange in the dispacher thread - so you need to at least provide an executor for the HTTP/1.1 server if you don't want to wedge when making parallel requests. But not the common FJP ;-)

Reply:
In a0859c7c68614e8c3f531b1d4d9357f3f48cf5ab,

- Used a dedicated executor (!= FJP) only for HTTP/1.1 servers (also documented the rationale in the sources)
- Added `SystemDiagnosticsCollector` to aid with troubleshooting

Verified that test passes on Windows, Linux, and macOS hosts with 30 repetitions.

Reply:
After consulting @jaikiran, removed `SystemDiagnosticsCollector` in ad73d38cd0382308224f721722367fd289fdc9d5. I will separately explore the possibilities of adding a similar troubleshooting tool in a more generic fashion.